From c762865aef6ad9ccb06503bce9f005fd3bfefaff Mon Sep 17 00:00:00 2001
From: Michael Benfield <mbenfield@google.com>
Date: Thu, 27 Jan 2022 20:03:44 +0000
Subject: [PATCH] fix bitwise or warning

On recent versions of LLVM this code triggers -Wbitwise-instead-of-logical

---
 media_driver/agnostic/common/codec/hal/codechal_decode_vc1.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/media_driver/agnostic/common/codec/hal/codechal_decode_vc1.cpp b/media_driver/agnostic/common/codec/hal/codechal_decode_vc1.cpp
index 64672f4d4..6426b0961 100644
--- a/media_driver/agnostic/common/codec/hal/codechal_decode_vc1.cpp
+++ b/media_driver/agnostic/common/codec/hal/codechal_decode_vc1.cpp
@@ -3222,7 +3222,7 @@ MOS_STATUS CodechalDecodeVc1::SetFrameStates()
     bool pictureIsI = m_mfxInterface->IsVc1IPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) ? true : false;
     bool pictureIsP = m_mfxInterface->IsVc1PPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) ? true : false;
     bool pictureIsB =
-        (m_mfxInterface->IsVc1BPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) |
+        (m_mfxInterface->IsVc1BPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType) ||
             m_mfxInterface->IsVc1BIPicture(currPic, m_vc1PicParams->picture_fields.is_first_field, picType))
             ? true
             : false;
-- 
2.31.0

